src/sql/workbench/contrib/queryplan2/browser/graphElementPropertiesView.ts

```diff
@@ -0,0 +1,260 @@
+import { Table } from 'sql/base/browser/ui/table/table';
+import { RESULTS_GRID_DEFAULTS } from 'sql/workbench/common/constants';
+import { isString } from 'vs/base/common/types';
+import { ActionBar } from 'sql/base/browser/ui/taskbar/actionbar';
+import { ActionsOrientation } from 'vs/base/browser/ui/actionbar/actionbar';
+
+
+export class GraphElementPropertiesView {
+
+	// Title bar with close button action
+	private _propertiesTitle!: HTMLElement;
+	private _titleText!: HTMLElement;
+	private _titleActionBarContainer!: HTMLElement;
+	private _titleActionBar: ActionBar;
+
+	// Div that holds the name of the element selected
+	private _operationName!: HTMLElement;
+
+	// Action bar that contains sorting option for the table
+	private _tableActionBarContainer!: HTMLElement;
+	private _tableActionBar!: ActionBar;
+
+	// Properties table
+	private _table: Table<any>;
+	private _dataView: TableDataView<Slick.SlickData>;
+	private _data: { [key: string]: string }[];
+	private _tableContainer!: HTMLElement;
+	private _actualTable!: HTMLElement;
+
+	// Table dimensions.
+	private _tableWidth = 485;
+	private _tableHeight = 420;
+
+	public constructor(
+		private _parentContainer: HTMLElement,
+		private _themeService: IThemeService,
+		private _model: GraphElementPropertyViewData = <GraphElementPropertyViewData>{}
+	) {
+		this._parentContainer.style.display = 'none';
+
+		this._propertiesTitle = DOM.$('.title');
+		this._parentContainer.appendChild(this._propertiesTitle);
+
+		this._titleText = DOM.$('h3');
+		this._titleText.classList.add('text');
+		this._titleText.innerText = localize('nodePropertyViewTitle', "Properties");
+		this._propertiesTitle.appendChild(this._titleText);
+
+		this._titleActionBarContainer = DOM.$('.action-bar');
+		this._propertiesTitle.appendChild(this._titleActionBarContainer);
+		this._titleActionBar = new ActionBar(this._titleActionBarContainer, {
+			orientation: ActionsOrientation.HORIZONTAL, context: this
+		});
+		this._titleActionBar.pushAction([new ClosePropertyViewAction()], { icon: true, label: false });
+
+		this._operationName = DOM.$('h3');
+		this._operationName.classList.add('operation-name');
+		this._parentContainer.appendChild(this._operationName);
+
+		this._tableActionBarContainer = DOM.$('.table-action-bar');
+		this._parentContainer.appendChild(this._tableActionBarContainer);
+		this._tableActionBar = new ActionBar(this._tableActionBarContainer, {
+			orientation: ActionsOrientation.HORIZONTAL, context: this
+		});
+		this._tableActionBar.pushAction([new SortPropertiesByDisplayOrderAction(), new SortPropertiesAlphabeticallyAction()], { icon: true, label: false });
+
+
+		this._tableContainer = DOM.$('.table-container');
+		this._parentContainer.appendChild(this._tableContainer);
+
+		this._actualTable = DOM.$('.table');
+		this._tableContainer.appendChild(this._actualTable);
+
+		this._dataView = new TableDataView();
+		this._data = [];
+
+		const columns: Slick.Column<any>[] = [
```

Comment:
You should give this an actual type (that extends from Slick.SlickData). If you're adding custom props you can just define that in this module.